lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4341,6 +4342,18 @@ namespace {
     }
 
     Decl *VisitFieldDecl(const clang::FieldDecl *decl) {
+      if (decl->hasAttr<clang::NoUniqueAddressAttr>()) {
+        if (const auto *rd = decl->getType()->getAsRecordDecl()) {
```

Comment:
Does this account for the scenario where the type is typedef'd?

Reply:
It seems to work, added test.